instrumentation/http-url-connection/src/main/java/io/opentelemetry/auto/instrumentation/httpurlconnection/HttpUrlConnectionTracer.java

```diff
@@ -53,4 +51,14 @@ protected String requestHeader(HttpURLConnection httpURLConnection, String name)
   protected String responseHeader(Integer integer, String name) {
     return null;
   }
+
+  @Override
+  protected String getInstrumentationName() {
+    return "io.opentelemetry.auto.http-url-connection";
+  }
+
+  @Override
+  public Span startSpan(String spanName) {
```

Comment:
Consider adding a comment when this should be used vs the one from the parent.

Reply:
> I have review only part of all changed files. This PR is a great step forward in the right direction, thank you @heyams for doing it. But there is quite a lot of common code between existing Tracers from the "server branch" and ones introduced here. We have to consolidate that into proper hierarchy.
> 
> Now, the question is should we do that consolidation in this PR or in a follow up. I have a strong opinion about the end result should look like and I will enjoy cleaning up the hierarchy. So I will be happy to do that follow up job. But if you, @heyams , want to play with it, be my guest and do it right here :)

I can refactor it further in another follow up PR.  My goal is to finish all the decorators in a 2nd PR, then refactor the common code between server and client.  I can continue with this path since I am familiar with the code now.  Good point.